### ISSUE TITLE
[Heartbeat] Client doesn't get revoked/wrong.host tls certificate status

### ISSUE THREAD

**Issue by shahzad31** [elastic/beats#18828]
In heartbeat client, we have TLS certificate check against http monitors, right now it covers expired and ok status. It will be very useful if it can also cover other invalid certificate status like 

![image](https://user-images.githubusercontent.com/3505601/83181266-ee78a500-a124-11ea-8b50-6ca338d42c40.png)

Further info about these status can be viewed at https://badssl.com/

**Comment by andrewvc** [comment]
Closed because cert revocation is largely broken and unused.

**Comment by shahzad31** [comment]
Offline comments:


> Many of those issues are not specifically cert issues. For instance, wrong host. There's nothing wrong with the cert, but someone's installed it on the wrong server. The regular heartbeat checks would catch that. Putting it on the certs page is a bit confusing IMHO

> WRT revocation, cert revocation is a huge minefield

> I'll ask the question, has anyone here ever encountered a revoked cert in the wild? I'd be willing to bet the answer is no, and there are some good reasons for that

> https://www.ssl.com/article/how-do-browsers-handle-revoked-ssl-tls-certificates/ is a good article on some of the issues with revocation
ssl.comssl.com
> How Do Browsers Handle Revoked SSL/TLS Certificates? - SSL.com
This article details current browser programs for checking the validity of SSL/TLS certificates, including a cross-browser test of revoked certificates.

> Let's encrypt in many ways was built because revocation is broken and hard to build in a reasonable way. That's why it has such short cert lifetimes

> TL;DR Revocation is a very rare and somewhat broken thing when it comes to TLS certs. Building it would be complex, and would provide very little value. (edited) 

> Here's another article on the many problems with the various methods of working with cert revocation

> https://medium.com/@alexeysamoshkin/how-ssl-certificate-revocation-is-broken-in-practice-af3b63b9cb3
MediumMedium
> SSL certificate revocation and how it is broken in practice
Explore certificate revocation solutions: CRL, OCSP, OCSP stapling, must staple. Check out server implementation issues and browser support

> Let's remember that our goal is always to provide the maximum value to users, and not to do "fill in the blanks" feature development. Our goal is not to check for possible cert related issue, but to help customers with the problems they actually have.

> WRT the untrusted root error, the regular heartbeat check will fail there as well. It's also hard to say what an untrusted root means. It doesn't necessarily mean the cert is bad, it could just mean heartbeat hasn't been setup with that root
